regulations/static/regulations/js/unittests/specs/views/comment-view-spec.js

```diff
@@ -0,0 +1,108 @@
+var chai = require('chai');
+var expect = chai.expect;
+var sinon = require('sinon');
+var jsdom = require('mocha-jsdom');
+var localStorage = require('node-localstorage');
+
+var storage = new localStorage.LocalStorage('.');
+
+describe('CommentView', function() {
+  jsdom();
+
+  var $el, commentView, CommentView, CommentEvents;
+
+  before(function() {
+    Backbone = require('backbone');
+    $ = require('jquery');
+    Backbone.$ = $;
+    CommentView = require('../../../source/views/comment-view');
+    CommentEvents = require('../../../source/events/comment-events');
+    window.localStorage = storage;
+  });
+
+  beforeEach(function() {
+    $el = $(
+      '<div class="comment">' +
+        '<div class="comment-clear">Clear</div>' +
+        '<form>' +
+          '<div class="editor-container"></div>' +
+          '<div class="queued"></div>' +
+          '<input type="file">' +
+          '<button type="submit">Save</button>' +
+          '<div class="status"></div>' +
+        '</form>' +
+      '</div>'
+    );
+    $(document.body).empty().append($el);
+    commentView = new CommentView({el: $el});
+  });
+
+  it('setSection()', function() {
+    commentView.setSection('2016_02749-III-B');
+    expect(commentView.section).to.equal('2016_02749-III-B');
+    expect(commentView.key).to.equal('comment:2016_02749-III-B');
+  });
+
+  it('fetches from localstorage', function() {
+    var payload = {
+      comment: 'like',
+      files: [
+        {key: '12345', name: 'attachment.txt'}
+      ],
+    };
+    commentView.setSection('2016_02479');
+    window.localStorage.setItem('comment:2016_02479', JSON.stringify(payload));
+    expect(commentView.getStorage()).to.deep.equal(payload);
+  });
+
+  it('writes to localstorage', function() {
+    commentView.setSection('2016_02479');
+    sinon.stub(commentView.editor, 'getContent').returns('dislike');
+    commentView.addQueueItem('abcde', 'attachment.txt');
+    commentView.setStorage();
+    var payload = {
+      comment: 'dislike',
+      files: [
+        {key: 'abcde', name: 'attachment.txt'}
+      ],
+    };
+    expect(commentView.getStorage()).to.deep.equal(payload);
+  });
+
+  it('adds a queue item', function() {
+    commentView.addQueueItem('key', 'name');
+    var $item = commentView.$queued.find('.queue-item');
+    expect($item.length).to.equal(1);
+    expect($item.data('key')).to.equal('key');
```

Comment:
These are S3 keys, so they're long random strings. I'll split the difference and use short random strings.